src/App.js

```diff
@@ -97,19 +92,23 @@ const App = () => {
       }
 
       setExpenses([...expenses, newExpense]);
-      setExpenseWasAdded(true);
-      setShowExpenseAddedAlert(true);
-      closeExpenseAddedAlert();
+      expenseWasAdded = true;
     } catch (error) {
-      setExpenseWasAdded(false);
-      setShowExpenseAddedAlert(true);
-      closeExpenseAddedAlert();
+      expenseWasAdded = false;
     }
+
+    setActiveAlert(
+      <ExpenseAddedAlert
+        expenseWasAdded={expenseWasAdded}
+        setActiveAlert={setActiveAlert}
```

Comment:
couldn't it be nice if the Alert did not have to know about the activeAlert variable and only have a callback?

Reply:
Not sure if I understand. We don't actually pass `activeAlert` to the alert itself.

Reply:
yep, but the setter. we could just pass the callback implementation (similar to what we do in alert)